### DISCUSSION TITLE
剪刀 石頭 布!

### DISCUSSION THREAD

**Discussion by Chengbosco** [1101-GS4719-C/Python-Community-GS4719-C#1665] [Show and tell]
這是我在升高一時花了一小段時間自學的東西，當初想說程式還蠻有趣，就嘗試把剪刀石頭布用程式來表達了，條件是贏三次。
其中有用串列來記錄勝負次數，再用count判斷，也有用到f固定格式，還有基本的迴圈，還有布林值來判斷。
-------------------------------------
from random import choice
電腦=["剪刀","石頭","布"]
玩家=[]
玩家戰績=[]
電腦戰績=[]  
a=0
b=0
game=True
if 玩家戰績.count("贏") ==3:
    game=False
if 電腦戰績.count("贏") ==3:
    game=False
while game==True:
    r=(input("你要出什麼[剪刀,石頭,布]:"))
    print(f"玩家出{r}")
    w=choice(電腦)
    print(f"電腦出{w}")
    
    if w =="剪刀":
     if r =="布":
         print("你輸")
         玩家戰績.append("輸")
         電腦戰績.append("贏")
         b+=1
     elif r =="石頭":
         print("你贏")
         玩家戰績.append("贏")
         a+=1
         電腦戰績.append("輸")
     elif r =="剪刀":
         print("平手")
         玩家戰績.append("平")
         電腦戰績.append("平")
    if w =="石頭":
     if r =="布":
         print("你贏")
         玩家戰績.append("贏")
         a+=1
         電腦戰績.append("輸")
     elif r =="石頭":
         print("平手")
         玩家戰績.append("平")
         電腦戰績.append("平") 
     elif r =="剪刀":
         print("你輸")
         玩家戰績.append("輸")
         電腦戰績.append("贏")
         b+=1
    if w =="布":
      if r =="布":
         print("平手")
         玩家戰績.append("平")
         電腦戰績.append("平")
      elif r =="石頭":
         print("你輸")
         玩家戰績.append("輸")
         電腦戰績.append("贏")
         b+=1
      elif r =="剪刀":
         print("你贏")
         玩家戰績.append("贏")
         a+=1
         電腦戰績.append("輸")
    if 玩家戰績.count("贏") ==3:
        game=False
        print("you win")
    if 電腦戰績.count("贏") ==3:
        game=False
        print("you lose")
#print("你的戰績",玩家戰績)
#print("電腦的戰績",電腦戰績)
---------------------------------------------------
其實就只是窮舉法。

**Comment by Chengbosco** [comment]
當初有看到其他人先隨機電腦選一個，再去設另外兩個，比我簡單多了。{if 電腦選石頭:剪刀=0  布=1}
在整理資料的時候就在思考什麼是**好的程式**，可能就是占的空間少和處理速度快不快吧。

**Comment by 1chooo** [comment]
看了你的文章覺得很有趣，因此想用現在正夯的 ChatGPT 快速寫一個剪刀、石頭、布程式。

#### 以下是問答過程
Q: 可以幫我用 python 寫一個剪刀石頭布的遊戲，對象是跟電腦玩，只要贏三次就宣告勝利

A: 當然可以！以下是一個簡單的Python代碼示例，實現剪刀石頭布遊戲，您對電腦進行多輪比賽，第一方贏得三輪即為勝利。遊戲過程中，您可以輸入"剪刀"、"石頭"或"布"來選擇您的動作，電腦則會隨機選擇動作。
```python
import random

def get_user_choice():
    user_choice = input("請選擇剪刀、石頭或布：")
    return user_choice

def get_computer_choice():
    choices = ["剪刀", "石頭", "布"]
    computer_choice = random.choice(choices)
    return computer_choice

def determine_winner(user, computer):
    if user == computer:
        return "平局"
    if (user == "剪刀" and computer == "布") or (user == "石頭" and computer == "剪刀") or (user == "布" and computer == "石頭"):
        return "玩家勝利"
    return "電腦勝利"

user_wins = 0
computer_wins = 0

while user_wins < 3 and computer_wins < 3:
    user_choice = get_user_choice()
    computer_choice = get_computer_choice()

    print(f"玩家選擇：{user_choice}")
    print(f"電腦選擇：{computer_choice}")

    result = determine_winner(user_choice, computer_choice)
    print(f"結果：{result}\n")

    if result == "玩家勝利":
        user_wins += 1
    elif result == "電腦勝利":
        computer_wins += 1

if user_wins >= 3:
    print("玩家獲勝！")
else:
    print("電腦獲勝！")
```

這個代碼包含了用戶和電腦的選擇、贏家的確定，以及遊戲循環，直到有一方贏得三輪為止。希望這對你有所幫助！
